index.html

```diff
@@ -1,88 +1,61 @@
-            <h1>Vaccine Availability Status</h1>
-            <h4 id="refreshTimerText"></h4>
-            <table class="input">
-                <tbody>
-                    <tr>
-                        <td>Date [DD-MM-YYYY]</td>
-                        <td><input type="text" id="dateForCheckingAvailability" /></td>
-
-                    </tr>
-
-                    <tr>
-                        <td>Disctrict Code(s) [Comma Separated]</td>
-                        <td><input type="text" id="disctrictCodes" /></td>
-                        </td>
-                    </tr>
-                    <tr>
-                        <td id="rightAlign">Press Enter to Search:</td>
-                        <td>
-                            <center> <button type="button" onclick="search();">Search</button> </center>
-                        </td>
-                    </tr>
-                </tbody>
-            </table>
-            <p>
-                Options</br>
-                <input type="checkbox" id="optionsAge18plus" checked>Age 18+ &amp &lt 45 </input></br>
-                <input type="checkbox" id="optionsAge45plus">Age 45+ </input> </br>
-                <input type="checkbox" id="optionsDose1" checked>Dose 1 </input></br>
-                <input type="checkbox" id="optionsDose2">Dose 2 </input>
-            </p>
-            <h4 id="API_CALL_Count"></h4>
-
-
-            <table class="output" id="mainTable">
-                <tbody>
-                    <tr>
-                        <td>District Code</td>
-                        <td>District Name</td>
-                        <td>Availability</td>
-                        <td>Name</td>
-                        <td>Pin Code</td>
-                        <td>Available Capacity for Dose 1</td>
-                        <td>Available Capacity for Dose 2</td>
-                    </tr>
-                </tbody>
-            </table>
-        </div>
+      <div class="center">
+        <h1>Vaccine Availability Status</h1>
+        <h4 id="refreshTimerText"></h4>
+        <table class="input">
+          <tbody>
+            <tr>
+              <td>Date [DD-MM-YYYY]</td>
+              <td><input type="text" id="dateForCheckingAvailability" /></td>
+            </tr>
+
+            <tr>
+              <td>Disctrict Code</td>
+              <td><input type="text" id="districtCodes" value="392, 395" /></td>
+            </tr>
+          </tbody>
+        </table>
+        <button type="button" id="btn" onclick="main()">Search</button>
+
+        <p>
+          Options<br />
+          <input type="checkbox" id="optionsAge18plus" checked />Age 18+<br />
+          <input type="checkbox" id="optionsAge45plus" />Age 45+<br />
+          <input type="checkbox" id="optionsDose1" checked />Dose 1<br />
+          <input type="checkbox" id="optionsDose2" />Dose 2
+        </p>
+        <h4 id="API_CALL_Count"></h4>
+
+        <table class="output" id="mainTable">
+          <thead>
```

Comment:
There is a function called "search()" in app.js where in all records in this Table with ID mainTable are cleared before new records are added on a refresh.
It relies on a logic to clear all rows except the 1st one. [Line number 155-160 in app.js file submitted in this PR]
```
  if (table.rows.length > 1) {
    console.log(table.rows.length)
    for (i = -1 * table.rows.length; i < -1; i++) {
      table.deleteRow(i * -1 - 1)
    }
  }
```


The code changes submitted here in index.html correctly moves the first row from the table body to table header.
However, I would like to confirm if you tested, whether the table clears correctly without error with the existing logic above when the 1st row (that acted as header for the table) is shifted from table body to table header.

Reply:
I refactored the table to include a `<thead>` and a `<tbody>`. This is for accessibility reasons, but it also makes it much easier to delete the data without touching the headers. By just overriding the whole `<tbody>` with an empty one, it clears it out entirely. It also has the benefit of being extremely efficient, no matter how many rows of data are there (because it's just deleting a single element rather than looping through a variable number of elements). Tested working in as many cases as I can think of, but can't guarantee it without unit tests.